### PR TITLE
Add documentation about replica retries.

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This is related to #59263.

### CODE REVIEW

**Comment by tbrooks8** [comment]
I opened this documentation PR to see what we think about this organization.

File: docs/reference/index-modules/indexing-pressure.asciidoc [hunk 1]
```diff
@@ -54,6 +54,20 @@ overhead (buffers, listeners, etc) which also require heap space. Other
...
+
+When a replica is unable to process a new indexing request, the replica is marked
+as failed by Elasticsearch. In order to minimize unnecessary replica failures in
+the face of transient failures (memory limits, circuit breaking, network
+disconnects, etc) the primary will retry replica requests. The retries will be
+attempted with exponential backoff until a configurable timeout is hit. Once the
+retries timeout, the replica will be failed.
+
+If the master chooses to fail the replica for other reasons during the retries, the
+retry attempts will be cancelled.
```
Comment (jrodewig) [inline docs/reference/index-modules/indexing-pressure.asciidoc:69]: Rewording to remove redundancy and avoid future tense.
```suggestion
To minimize replica failures due to transient issues such as memory limits,
circuit breaking, or network disconnects, primary shards retry unsuccessful
indexing requests sent to replicas. These retries continue with an exponential
backoff until a configurable timeout is hit. Once the timeout is hit, {es} marks
the replica as failed. If the replica fails for another reason during retries, the 
primary shard stops further retry attempts.
```

File: docs/reference/index-modules/indexing-pressure.asciidoc [hunk 2]
```diff
@@ -71,3 +85,14 @@ retrieve indexing pressure metrics.
...
   primary operations. When replica operations consume 1.5x this limit, the node
   will reject new replica operations. Defaults to 10% of the heap.
+
+[discrete]
+[[replica-resiliency-settings]]
+=== Replica resiliency settings
+
+`indices.replication.initial_retry_backoff_bound`::
+  The initial backoff bound for a replica retry. The first retry will be a
+  randomly generated period bound by this value. Defaults to 50ms.
```
Comment (jrodewig) [inline docs/reference/index-modules/indexing-pressure.asciidoc:95]:
```suggestion
  Initial backoff bound for a replica retry. The first retry is a
  randomly generated period, bound by this value. Defaults to `50ms`
  (50 milliseconds).
```

File: docs/reference/index-modules/indexing-pressure.asciidoc [hunk 3]
```diff
@@ -71,3 +85,14 @@ retrieve indexing pressure metrics.
...
+
+`indices.replication.retry_timeout`::
+  The timeout period for a replica retries. Defaults to 60s.
```
Comment (jrodewig) [inline docs/reference/index-modules/indexing-pressure.asciidoc:98]:
```suggestion
  Timeout period for a replica retries. Defaults to `60s` (60 seconds).
```

File: docs/reference/index-modules/indexing-pressure.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/index-modules/indexing-pressure.asciidoc:69]: It may be helpful to mention the settings by name here.

**Review by jrodewig — APPROVED** [review]
LGTM. I left some non-blocking suggestions which you can use or ignore as you see fit. Thanks!

File: docs/reference/index-modules/indexing-pressure.asciidoc [hunk 4]
```diff
@@ -54,6 +54,20 @@ overhead (buffers, listeners, etc) which also require heap space. Other
 components of {es} also require memory. Setting this limit too high can deny
 operating memory to other operations and components.
 
+[discrete]
+[[replica-resilience]]
+=== Replica resilience
+
+When a replica is unable to process a new indexing request, the replica is marked
```
Comment (ywelsch) [inline docs/reference/index-modules/indexing-pressure.asciidoc:61]: This suggest that the replica has received the indexing request. I wonder if this should be restated as "Replication resilience", i.e. the process of replicating the data from the primary to the replica.

File: docs/reference/index-modules/indexing-pressure.asciidoc [hunk 5]
```diff
@@ -71,3 +85,14 @@ retrieve indexing pressure metrics.
   this limit is reached or exceeded, the node will reject new coordinating and
   primary operations. When replica operations consume 1.5x this limit, the node
   will reject new replica operations. Defaults to 10% of the heap.
+
+[discrete]
+[[replica-resiliency-settings]]
+=== Replica resiliency settings
+
+`indices.replication.initial_retry_backoff_bound`::
+  The initial backoff bound for a replica retry. The first retry will be a
```
Comment (ywelsch) [inline docs/reference/index-modules/indexing-pressure.asciidoc:94]: replication retry

File: docs/reference/index-modules/indexing-pressure.asciidoc (hunk 3 above)
Comment (ywelsch) [inline docs/reference/index-modules/indexing-pressure.asciidoc:98]: id  here

**Comment by jrodewig** [comment]
Closing this due to inactivity. @Tim-Brooks please feel free to re-open if this is something you're still interested in tackling. Alternatively, I can create a docs request for my team to investigate. Thanks!

**Pull request closed without merging**